server/lib/routes/get-authorization.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const url = require('url');
+
+const logger = require('../logging/log')('server.get-authorization');
+
+const ACTION_TO_PATHNAMES = {
+  'email': '',
+  'force_auth': 'force_auth',
+  'signin': 'signin',
+  'signup': 'signup'
+};
+
+function actionToPathname(action) {
+  if (action === undefined) {
+    return '';
+  }
+
+  if (ACTION_TO_PATHNAMES.hasOwnProperty(action)) {
+    return ACTION_TO_PATHNAMES[action];
+  }
+
+  throw new Error('Bad action parameter');
+}
+
+module.exports = function () {
+  return {
+    method: 'get',
+    path: '/authorization',
+    process: function (req, res) {
+      let err = false;
+      const route = url.parse('/', true);
+
+      try {
+        route.pathname += actionToPathname(req.query.action);
+      } catch (e) {
+        err = true;
+        logger.error('Invalid request parameter', req.query.action);
```

Comment:
If the built in validation is leaned upon, [logging is already taken care of for you](https://github.com/mozilla/fxa-content-server/blob/eabe9c022d6135bd9ac10d11229b56a95ff3aa77/server/lib/routes.js#L117).